src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,106 @@
+package duke;
+
+import duke.command.Command;
+import duke.command.Parser;
+import duke.exception.DukeException;
+import duke.storage.Storage;
+import duke.task.TaskList;
+import duke.ui.SimpleCliUi;
+import duke.ui.Ui;
+
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+
+/**
+ * Represents a task list chatbot named Duke.
+ *
+ * @author Terence Chong Guang Jun
+ */
+public class Duke {
+    /**
+     * Specifies the date format used by <code>Duke</code> for receiving date input.
+     */
+    public static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("ddMMyyyy HHmm");
+
+    /**
+     * Specifies that <code>Duke</code> should continue to read the next command.
+     */
+    public static final int CODE_CONTINUE = 0;
+
+    /**
+     * Specifies that <code>Duke</code> should say bye and exit.
+     */
+    public static final int CODE_EXIT = 1;
+
+    private Ui ui;
+    private Storage storage;
+    private TaskList taskList;
+
+    /**
+     * Creates a new <code>Duke</code> instance. The specified file path is used in
+     * loading and saving the <code>TaskList</code> for this <code>Duke</code>.
+     *
+     * @param filePath Specifies the path where the <code>TaskList</code> is loaded from.
+     */
+    public Duke(String filePath) {
+        this(new SimpleCliUi(), new Storage(filePath));
```

Comment:
In this case I suppose the intention was so that someone creating a new `Duke` instance could choose to select which `filePath` they wanted to use, if they wanted to specify something rather than the default. Or even more specifically, they could supply the Ui and the Storage themselves.